pkg/storage/chunk/client/azure/blob_storage_client.go

```diff
@@ -53,7 +53,7 @@ var (
 
 	defaultAuthFunctions = authFunctions{
 		NewOAuthConfigFunc: adal.NewOAuthConfig,
-		NewServicePrincipalTokenFromFederatedTokenFunc: adal.NewServicePrincipalTokenFromFederatedToken,
+		NewServicePrincipalTokenFromFederatedTokenFunc: adal.NewServicePrincipalTokenFromFederatedToken, //nolint:staticcheck // SA1019: use of deprecated function.
```

Comment:
I've added this lint ignore, because the deprecated function still works. It could be replaced, but I suggest doing that in a separate PR by someone who is more familiar with this code.